pkg/reconciler/shared/tektoninstallerset/types.go

```diff
@@ -58,8 +61,31 @@ func newTisMetaWithGenerateName(namePrefix string) *tisMeta {
 	return tis
 }
 
-func (tis *tisMeta) config(ctx context.Context, ci ComponentInstaller) {
+func (tis *tisMeta) config(ctx context.Context, ci ComponentInstaller) error {
 	tis.Labels = ci.GetLabels(ctx)
 	tis.Annotations = ci.GetAnnotations(ctx)
 	tis.OwnerReferences = ci.GetOwnerReferences(ctx)
+
+	if err := validateMetadata(tis); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateMetadata(tis *tisMeta) error {
+	key := validatelabels(tis.Labels)
+	if key != "" {
+		return fmt.Errorf("Label with key: `%s` is not been added to create the installerset", key)
+	}
+	return nil
+}
+
+func validatelabels(labels map[string]string) string {
+	for _, keys := range requiredLabels {
+		if _, ok := labels[keys]; !ok {
```

Comment:
can we make it like checking all the keys, if it is not there append in some variable, return all the keys as an error which is not there in spite of single every time